src/esw/brushed_motors/Controller.cpp

```diff
@@ -76,15 +94,91 @@ void Controller::moveOpenLoop(float input) {
 
         I2C::transact(deviceAddress, motorIDRegMask | OPEN_PLUS_OP, OPEN_PLUS_WB,
                       OPEN_PLUS_RB, buffer, UINT8_POINTER_T(&angle));
-        currentAngle = (float) ((((float) angle / quadCPR) * 2 * M_PI) - M_PI);
+        currentAngle = (float) (((float) angle / quadCPR) * 2 * M_PI);
     } catch (IOFailure& e) {
         ROS_ERROR("moveOpenLoop failed on %s", name.c_str());
     }
 }
 
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: I2C bus, returns if the MCU is calibrated
+bool Controller::isCalibrated() {
+    uint8_t calibration_status;
+
+    try {
+        makeLive();
+        if (isControllerCalibrated) return true;
+
+        I2C::transact(deviceAddress, motorIDRegMask | IS_CALIBRATED_OP, IS_CALIBRATED_WB,
+                      IS_CALIBRATED_RB, nullptr, UINT8_POINTER_T(&calibration_status));
+
+        isControllerCalibrated = calibration_status;
+
+    } catch (IOFailure& e) {
+        ROS_ERROR("isCalibrated failed on %s", name.c_str());
+    }
+
+    return calibration_status;
+}
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: I2C bus, enables or disables limit switches
+void Controller::enableLimitSwitches(bool enable) {
+    try {
+        makeLive();
+
+        limitAEnable = limitAPresent && enable;
+
+        uint8_t buffer[1];
+        memcpy(buffer, UINT8_POINTER_T(&limitAEnable), sizeof(limitAEnable));
+        I2C::transact(deviceAddress, motorIDRegMask | ENABLE_LIMIT_A_OP, ENABLE_LIMIT_A_WB,
+                      ENABLE_LIMIT_A_RB, buffer, nullptr);
+
+        limitBEnable = limitBPresent && enable;
+
+        memcpy(buffer, UINT8_POINTER_T(&limitBEnable), sizeof(limitBEnable));
+        I2C::transact(deviceAddress, motorIDRegMask | ENABLE_LIMIT_B_OP, ENABLE_LIMIT_B_WB,
+                      ENABLE_LIMIT_B_RB, buffer, nullptr);
```

Comment:
Shouldn't we do transactions when limit switches are present? It may also be nice to have smaller try catch blocks just around transact statements but that's not super important